src/quartic/pipeline/validator/qdag.py

```
@@ -0,0 +1,41 @@
+from functools import reduce
+from quartic.pipeline.validator import dag_utils
+from quartic.common.exceptions import QuarticException
+import pprint
+
+class QuarticDag:
+    def __init__(self, nx_graph):
+        self.graph = nx_graph
+        if not dag_utils.check_dag(self.graph):
```

`check_dag` throws an exception, rather than returning a boolean.